From: Luming Yu <luyu@redhat.com>
Date: Mon, 24 Mar 2008 11:38:56 +0800
Subject: [ia64] remove assembler warnings on head.S
Message-id: 47E72250.5090908@redhat.com
O-Subject: [RHEL 5.2 PATCH] bz438230 : ia64: suspecious compile warning in brew
Bugzilla: 438230
RH-Acked-by: Prarit Bhargava <prarit@redhat.com>

bz438230

Description of problem:

In the build log diffs between -85.el5 and -86.el5, the following
warning popped up and looked interesting:

arch/ia64/kernel/head.S:###: Warning: Use of 'ldf.fill.nta' violates RAW
dependency 'CR[PTA]' (data)mm/mmap.
c: In function 'split_vma':

This is fixed by the attached patch, which has already been in the mainline kernel.

According to IPF SDM,Serialization is needed to ensure new values in PTA are
visible to the hardware Virtual Hash Page Table  (VHPT) walker before a
dependent instruction fetch or data access.

Upstream status:

091062284c05d13b3393f4fcfcedc0f52cb948b4

I have tested the patch. It fixes the problem.

Please review,and ACK.

Thanks,
Luming

 arch/ia64/kernel/head.S |    1 +
 1 files changed, 1 insertions(+), 0 deletions(-)

diff --git a/arch/ia64/kernel/head.S b/arch/ia64/kernel/head.S
index dded6f2..bd9d2da 100644
--- a/arch/ia64/kernel/head.S
+++ b/arch/ia64/kernel/head.S
@@ -1180,6 +1180,7 @@ tlb_purge_done:
 	RESTORE_REG(cr.dcr, r25, r17);;
 	RESTORE_REG(cr.iva, r25, r17);;
 	RESTORE_REG(cr.pta, r25, r17);;
+	srlz.d;;	// required not to violate RAW dependency
 	RESTORE_REG(cr.itv, r25, r17);;
 	RESTORE_REG(cr.pmv, r25, r17);;
 	RESTORE_REG(cr.cmcv, r25, r17);;
